apps/wallet/src/ui/app/hooks/useGasBudgetEstimationUnits.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type CoinStruct } from '@mysten/sui.js';
+
+import { DEFAULT_GAS_BUDGET_FOR_PAY } from '_redux/slices/sui-objects/Coin';
+
+// This is a helper function to get a set of gas coins that can cover a given amount
+// It is from Coins.selectCoinSetWithCombinedBalanceGreaterThanOrEqual from sui.js
+function getCoinSetWithCombinedBalanceGreaterThanOrEqual(
+    coins: CoinStruct[],
+    amount: bigint
+) {
+    // Sort coin by balance in an descending order
+    const sortedCoins = coins
+        .filter(({ lockedUntilEpoch }) => !lockedUntilEpoch)
+        .sort((a, b) => b.balance - a.balance);
+
+    // calculate total balance
+    const total = sortedCoins.reduce(
+        (acc, { balance }) => acc + BigInt(balance),
+        BigInt(0)
+    );
+
+    if (total < amount) {
+        return [];
+    } else if (total === amount) {
+        return sortedCoins;
+    }
+
+    let sum = BigInt(0);
+    const ret = [];
+    while (sum < total) {
+        // prefer to add a coin with smallest sufficient balance
+        const target = amount - sum;
+        const coinWithSmallestSufficientBalance = sortedCoins.find(
+            ({ balance }) => balance >= target
+        );
+        if (coinWithSmallestSufficientBalance) {
+            ret.push(coinWithSmallestSufficientBalance);
+            break;
+        }
+
+        const coinWithLargestBalance = sortedCoins.pop()!;
+        ret.push(coinWithLargestBalance);
+        sum += BigInt(coinWithLargestBalance.balance);
+    }
+    // sort coins by balance in descending order
+    return ret.sort((a, b) => a.balance - b.balance);
+}
+
+export function useGasBudgetEstimationUnits(
```

Comment:
this doesn't look like it needs to be a hook, could we move this out of the `/hooks` folder and re-name it accordingly?